### ISSUE TITLE
suspicious_arithmetic_impl and OpAssign

### ISSUE THREAD

**Issue by l0calh05t** [rust-lang/rust-clippy#5512]
While the `suspicious_arithmetic_impl` is a bit overeager itself (see #3215), it also fails to detect the type of mistake it is supposed to detect when an Op-trait is implemented in terms of the corresponding OpAssign-trait. In the example below, `Add` has "accidentally" been implemented in terms of `*=`/`MulAssign`, but clippy 0.0.212 (4ee12063 2020-02-01) does not issue a warning:

```rust
#[derive(Clone, Debug)]
struct Foo(f32);

impl std::ops::Mul<&Foo> for &Foo {
	type Output = Foo;
	fn mul(self, other: &Foo) -> Foo {
		let mut ret = self.clone();
		ret.0 *= other.0;
		ret
	}
}

impl std::ops::Add<&Foo> for &Foo {
	type Output = Foo;
	fn add(self, other: &Foo) -> Foo {
		let mut ret = self.clone();
		ret.0 *= other.0; // oops, but no warning
		ret
	}
}

fn main() {
	println!("{:?}", &Foo(2.0) * &Foo(3.0));
	println!("{:?}", &Foo(2.0) + &Foo(3.0));
}
```

**Comment by ThibsG** [comment]
This lint seems to be correctly triggered or am I missing something?
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a5dce40da6511b2d4f6544a8e38f2d4f)

**Comment by l0calh05t** [comment]
@ThibsG Doesn't trigger on your playground link either. Tried both debug and release and stable and nightly.

**Comment by ThibsG** [comment]
But if you do "Tools" then "Clippy" it triggers right?

**Comment by l0calh05t** [comment]
It does, yes. Oddly, it does for me locally now to... closing for now.
